__tests__/app.test.js

```diff
@@ -337,3 +337,36 @@ describe("GET /api/users", () => {
       });
   });
 });
+
+describe("GET /api/articles?topic", () => {
+  it("responds with a 200 status message", () => {
+    return request(app).get("/api/articles?topic=mitch").expect(200);
+  });
+  it("responds with an an array of object which contains article information", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .then(({ body }) => {
+        expect(body.length).toBe(12);
+        body.forEach((article) => {
+          expect(article).toMatchObject({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: "mitch",
+            author: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(String),
+          });
+        });
+      });
+  });
+  it("responds with a 404 and a message when the topic is not found", () => {
+    return request(app)
+      .get("/api/articles?topic=PossiblyNotATopic")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("Topic Not Found");
+      });
+  });
+});
```

Comment:
Nice one - we're very close. I think we could consider one more test scenario:
- Valid 200 request for a topic query that exists but has no articles associated with it. 

We will need to consider how we can differentiate between an empty response from the query being due to the topic not existing or because there's no articles with that valid topic.

Hint: Have a look at the notes on the checkExists function to help you with this.